### PR TITLE
Add `gfortranImpilicitArgument` flag in integration tests

### PR DESCRIPTION
**Pull request by Pranavchiku**
[1942#pullrequestreview](https://github.com/lfortran/lfortran/pull/1942#pullrequestreview-1520393946)

### CODE REVIEW

**Comment by Pranavchiku** [comment]
I am not sure if this is the correct way

File: integration_tests/CMakeLists.txt [hunk 1]
```diff
@@ -110,9 +112,15 @@ macro(RUN_UTIL RUN_FAIL RUN_NAME RUN_FILE_NAME RUN_LABELS RUN_EXTRAFILES RUN_EXT
...
-            add_test(${name} ${CURRENT_BINARY_DIR}/${name})
+            if ("gfortranImplicitArgument" IN_LIST labels)
+                execute_process(COMMAND gfortran ${CMAKE_CURRENT_SOURCE_DIR}/${file_name}.f90
+				    ${extra_args} -fallow-argument-mismatch -o ${name})
+                add_test(${name} ${CURRENT_BINARY_DIR}/${name})
+            else ()
+                add_executable(${name} ${file_name}.f90 ${extra_files})
+                target_compile_options(${name} PUBLIC ${extra_args})
+                add_test(${name} ${CURRENT_BINARY_DIR}/${name})
+            endif()
```
Comment (certik) [inline integration_tests/CMakeLists.txt:123]: This wouldn't work?
```suggestion
            if ("gfortranImplicitArgument" IN_LIST labels)
                set(extra_args "${extra_args} -fallow-argument-mismatch")
            endif()
            add_executable(${name} ${file_name}.f90 ${extra_files})
            target_compile_options(${name} PUBLIC ${extra_args})
            add_test(${name} ${CURRENT_BINARY_DIR}/${name})
```

**Comment by certik** [comment]
Also you need to enable this test at the CI --- to run both `gfortran` and `gfortranImplicit...`.

File: integration_tests/CMakeLists.txt (hunk 1 above)
Comment (Pranavchiku) [reply integration_tests/CMakeLists.txt:123]: I tried this, it is not working for me not sure why.

Comment (certik) [reply integration_tests/CMakeLists.txt:123]: Ok, let's leave the current way.

Comment (Pranavchiku) [reply integration_tests/CMakeLists.txt:123]: Okay

Comment (Pranavchiku) [reply integration_tests/CMakeLists.txt:123]: It worked with `set(extra_args ${extra_args} -fallow-argument-mismatch)`

**Comment by certik** [comment]
I think that this is good. Let's enable it at the CI as well (just add it together with the regular `gfortran` tests).

**Comment by Pranavchiku** [comment]
I have done that in this PR itself.

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -549,6 +549,7 @@ jobs:
         run: |
             cd integration_tests
             ./run_tests.py -b gfortran
+            ./run_tests.py -b gfortranImplicitArgument
```
Comment (certik) [inline .github/workflows/CI.yml:552]:
```suggestion
            ./run_tests.py -b gfortran gfortranImplicitArgument
```

**Comment by certik** [comment]
Perfect. I left a suggestion above. Can you also please resolve the conflicts? Let's see that all tests run. Otherwise I think this is ready.

**Comment by Pranavchiku** [comment]
On it.

**Comment by certik** [comment]
For some reason it ran all 642 tests for the implicit argument thing, while I would expect to only run 1.

**Comment by Pranavchiku** [comment]
This is because whenever we are encountering "gfortran" we do `set(ADD_TEST ON)` irrespective of backend.

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -548,7 +548,7 @@ jobs:
         shell: bash -e -l {0}
         run: |
             cd integration_tests
-            ./run_tests.py -b gfortran
+            ./run_tests.py -b gfortran gfortranImplicitArgument
```
Comment (certik) [inline .github/workflows/CI.yml:551]: Remove

File: integration_tests/run_tests.py [hunk 4]
```diff
@@ -7,7 +7,7 @@
 # Initialization
 NO_OF_THREADS = 8 # default no of threads is 8
 SUPPORTED_BACKENDS = ['llvm', 'llvm2', 'llvm_rtlib', 'c',
-                      'cpp', 'x86', 'wasm', 'gfortran', 'llvmImplicit', 'llvmImplicitArgument']
+                      'cpp', 'x86', 'wasm', 'gfortran', 'llvmImplicit', 'llvmImplicitArgument', 'gfortranImplicitArgument']
```
Comment (certik) [inline integration_tests/run_tests.py:10]: remove

File: integration_tests/run_tests.py [hunk 5]
```diff
@@ -27,6 +27,9 @@ def run_test(backend):
     if backend == "gfortran":
         run_cmd(f"FC=gfortran cmake" + common,
                 cwd=cwd)
+    elif backend == "gfortranImplicitArgument":
+        run_cmd(f"FC=gfortran cmake" + common,
+                cwd=cwd)
```
Comment (certik) [inline integration_tests/run_tests.py:32]: Remove

**Review by certik — APPROVED** [review]
I think this works!

**Pull request merged**